etl/src/store/schema/base.rs

```diff
@@ -0,0 +1,30 @@
+use postgres::schema::{TableId, TableSchema};
+use std::sync::Arc;
+
+use crate::error::EtlResult;
+
+pub trait SchemaStore {
+    /// Returns table schema for table with id `table_id` from the cache.
+    ///
+    /// Does not load any new data into the cache.
+    fn get_table_schema(
+        &self,
+        table_id: &TableId,
+    ) -> impl Future<Output = EtlResult<Option<Arc<TableSchema>>>> + Send;
```

Comment:
Since `TableSchema` is immutable, we can just `Arc` it to avoid unnecessary clones.